### ISSUE TITLE
5.6 Branch ITs (around settings) are Broken

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#8331]
https://travis-ci.org/elastic/logstash/jobs/277515102

```sh
Failures:
  1) Test Logstash instance whose default settings are overridden start on a different HTTP port
     Failure/Error: expect(File.exists?(@logstash_default_logs)).to be true
     
       expected true
            got false
     # ./specs/settings_spec.rb:140:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
  2) Test Logstash instance whose default settings are overridden should write logs to a new dir
     Failure/Error: expect(File.exists?("#{temp_dir}/logstash-plain.log")).to be true
     
       expected true
            got false
     # ./specs/settings_spec.rb:64:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
  3) Test Monitoring API can configure logging
     Failure/Error: expect(logstash_service.monitoring_api.logging_get["loggers"]["logstash.agent"]).to eq ("DEBUG")
     
       expected: "DEBUG"
            got: nil
     
       (compared using ==)
     # ./specs/monitoring_api_spec.rb:121:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.23/lib/stud/try.rb:79:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.23/lib/stud/try.rb:95:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.23/lib/stud/try.rb:91:in `try'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/stud-0.0.23/lib/stud/try.rb:123:in `try'
     # ./specs/monitoring_api_spec.rb:105:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
  4) Test Logstash Slowlog should write logs to a new dir
     Failure/Error: expect(File.exists?(slowlog_file)).to be true
     
       expected true
            got false
     # ./specs/slowlog_spec.rb:43:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
Finished in 17 minutes 10 seconds (files took 4.59 seconds to load)
37 examples, 4 failures
Failed examples:
rspec ./specs/settings_spec.rb:125 # Test Logstash instance whose default settings are overridden start on a different HTTP port
rspec ./specs/settings_spec.rb:56 # Test Logstash instance whose default settings are overridden should write logs to a new dir
rspec ./specs/monitoring_api_spec.rb:100 # Test Monitoring API can configure logging
rspec ./specs/slowlog_spec.rb:33 # Test Logstash Slowlog should write logs to a new dir
```

**Comment by original-brownbear** [comment]
This may be easier for someone else to handle, there's quite a few changes here between `master` and `5.6` and I don't have a good handle on how the config logic worked then.

**Comment by jakelandis** [comment]
This is actual regression.  Will be fixed on #8337

**Comment by jakelandis** [comment]
fixed with https://github.com/elastic/logstash/pull/8339
